src/Report/Lcov.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of phpunit/php-code-coverage.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace SebastianBergmann\CodeCoverage\Report;
+
+use function count;
+use function dirname;
+use function file_put_contents;
+use function ksort;
+use function max;
+use function range;
+use SebastianBergmann\CodeCoverage\CodeCoverage;
+use SebastianBergmann\CodeCoverage\Directory;
+use SebastianBergmann\CodeCoverage\Driver\WriteOperationFailedException;
+use SebastianBergmann\CodeCoverage\Node\File;
+
+/**
+ * Ref: http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php.
+ */
+final class Lcov
+{
+    /**
+     * @throws WriteOperationFailedException
+     */
+    public function process(CodeCoverage $coverage, ?string $target = null, ?string $name = null): string
+    {
+        $lcovLines = [];
+        $report    = $coverage->getReport();
+
+        foreach ($report as $file) {
+            $lcovLines[] = 'TN:';
+            $lcovLines[] = 'SF:' . $file->pathAsString();
+            $lcovLines[] = 'FNF:' . $file->numberOfMethods();
+            $lcovLines[] = 'FNH:' . $file->numberOfTestedMethods();
+
```

Comment:
When running under Xdebug, I think there should be the raw data available to add `BRDA`/`BRF`/`BRH` segments with branch coverage data? See the text report (https://github.com/sebastianbergmann/php-code-coverage/blob/master/src/Report/Text.php#L74) for determining whether the data is available.

Reply:
The data `BRDA`/`BRF/`BRH` has been added to the latest commit. Cheers :+1: